### PR TITLE
Add end-to-end deregistration tests for Host Details page

### PR DESCRIPTION
**Pull request by jamie-suse**
# Description

Adds end-to-end deregistration tests for the Host Details page.

## How was this tested?

By adding end-to-end deregistration tests for the Host Details page.

### CODE REVIEW

File: test/e2e/cypress/e2e/host_details.cy.js [hunk 1]
```diff
@@ -355,4 +355,41 @@ context('Host Details', () => {
...
+      });
+    });
+
+    describe('"Clean up" button should deregister a host', () => {
+      before(() => {
+        cy.task('stopAgentsHeartbeat');
+      });
+
+      it('should allow to deregister a host after clean-up confirmation', () => {
+        cy.contains('button', 'Clean up', { timeout: 15000 }).click();
```
Comment (jamie-suse) [inline test/e2e/cypress/e2e/host_details.cy.js:378]: The value of `timeout` depends on Elixir environment variable [`config :trento, deregistration_debounce`](https://github.com/trento-project/web/blob/main/config/test.exs#L49). These timeouts are also present in the [Hosts List tests](https://github.com/trento-project/web/blob/main/test/e2e/cypress/e2e/hosts_overview.cy.js#L205) for the same reason. Could we abstract this to an environment variable somehow?

File: test/e2e/cypress/e2e/host_details.cy.js [hunk 2]
```diff
@@ -355,4 +355,41 @@ context('Host Details', () => {
       cy.get('span').find('svg').should('exist');
     });
   });
+
+  describe('Deregistration', () => {
+    describe('"Clean up" button should be visible only for an unhealthy host', () => {
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply test/e2e/cypress/e2e/host_details.cy.js:360]: Yes, let's have them!

File: test/e2e/cypress/e2e/host_details.cy.js [hunk 3]
```diff
@@ -355,4 +355,41 @@ context('Host Details', () => {
...
+      it('should not display the "Clean up" button for healthy host', () => {
+        cy.task('startAgentHeartbeat', [selectedHost.id]);
+        cy.contains('button', 'Clean up').should('not.exist');
```
Comment (arbulu89) [inline test/e2e/cypress/e2e/host_details.cy.js:363]: Nice! Actually I prefer this method over the one @rtorrero implemented in the host list view hehe
But to keep the same, we should still use other way around. Ruben added a `data-testid` so, let's use it I guess.
What do you think @rtorrero ?

File: test/e2e/cypress/e2e/host_details.cy.js [hunk 4]
```diff
@@ -355,4 +355,41 @@ context('Host Details', () => {
...
+        cy.get('@modal')
+          .find('.w-full')
+          .should(
+            'contain.text',
+            `Clean up data discovered by agent on host ${selectedHost.hostName}`
+          );
+        cy.get('@modal').contains('button', 'Clean up').click();
+
+        cy.get('@modal').should('not.exist');
+        cy.url().should('eq', cy.config().baseUrl + '/hosts');
```
Comment (arbulu89) [inline test/e2e/cypress/e2e/host_details.cy.js:391]: We should check that the row in the host list view for this host goes away

**Review by arbulu89 — COMMENTED** [review]
@jamie-suse I see that we use a totally different way compared with the HostList view. I think we should be consistent, and after the other is merged, follow the same logic.
@rtorrero Could you assist on this? And review as you have the best insights!

File: test/e2e/cypress/e2e/host_details.cy.js [hunk 5]
```diff
@@ -355,4 +355,42 @@ context('Host Details', () => {
...
+
+    describe('"Clean up" button should deregister a host', () => {
+      before(() => {
+        cy.task('stopAgentsHeartbeat');
+      });
+
+      it('should allow to deregister a host after clean-up confirmation', () => {
+        cy.contains('button', 'Clean up', { timeout: 15000 }).click();
+
+        cy.get('#headlessui-portal-root').as('modal');
```
Comment (rtorrero) [inline test/e2e/cypress/e2e/host_details.cy.js:380]: Regarding my previous comment: I think this is how you solved the `contains` getting the `Clean up` button from the modal. If that's so, then I'm up for it. I'll create a tech debt to update the other deregistration e2e tests.

**Review by rtorrero — APPROVED** [review]
LGTM now @jamie-suse , thanks!

**Review by arbulu89 — APPROVED** [review]

**Pull request merged**